service-base/src/main/java/org/eclipse/hono/service/credentials/CredentialsService.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2016, 2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ */
+
+package org.eclipse.hono.service.credentials;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.Verticle;
+import io.vertx.core.json.JsonObject;
+import org.eclipse.hono.util.CredentialsResult;
+import org.eclipse.hono.util.RegistrationResult;
+
+import java.net.HttpURLConnection;
+
+/**
+ * A service for keeping record of credentials for devices.
+ *
+ */
+public interface CredentialsService extends Verticle {
+
+    /**
+     * Gets credentials data by authId of a device ID with a specific type.
+     *
+     * @param tenantId The tenant the device belongs to.
+     * @param type The type of credentials to get.
+     * @param authId The authID of the device to get credentials data for.
+     * @param resultHandler The handler to invoke with the result of the operation. If credentials with the
+     *         given authID and type are registered for the tenant, the <em>status</em> will be {@link HttpURLConnection#HTTP_OK}
+     *         and the <em>payload</em> will contain the details of the credentials.
+     *         Otherwise the status will be {@link HttpURLConnection#HTTP_NOT_FOUND}.
+     */
+    void getCredentials(String tenantId, String type, String authId, Handler<AsyncResult<CredentialsResult>> resultHandler);
```

Comment:
Good point, I will add a @throws NullPointerException for all params.
We haven't specified that the get operation may not fully provide the three params, so it seems appropriate.